tree/ntupleutil/v7/inc/ROOT/RNTupleImporter.hxx

```diff
@@ -172,44 +169,33 @@ private:
       ~RImportGuard() { fImporter.ResetSchema(); }
    };
 
-   /// Leaf count arrays require special treatment. They are translated into RNTuple untyped collections.
+   /// Leaf count arrays require special treatment. They are translated into untyped vectors of untyped records.
    /// This class does the bookkeeping of the sub-schema for these collections.
    struct RImportLeafCountCollection {
       RImportLeafCountCollection() = default;
       RImportLeafCountCollection(const RImportLeafCountCollection &other) = delete;
       RImportLeafCountCollection(RImportLeafCountCollection &&other) = default;
       RImportLeafCountCollection &operator=(const RImportLeafCountCollection &other) = delete;
       RImportLeafCountCollection &operator=(RImportLeafCountCollection &&other) = default;
-      std::unique_ptr<RNTupleModel> fCollectionModel;             ///< The model for the collection itself
-      std::shared_ptr<RNTupleCollectionWriter> fCollectionWriter; ///< Used to fill the collection elements per event
-      std::unique_ptr<REntry> fCollectionEntry; ///< Keeps the memory location of the collection members
+      std::string fFieldName; ///< name of the untyped collection, e.g. `_collection0`, `_collection1`, etc.
+      Int_t fMaxLength = 0;   ///< Stores count leaf GetMaximum() to create large enough buffers for the array leafs
```

Comment:
Is there a reason to use `Int_t` rather than `std::int32_t` as we tend to do in the rest of the RNTuple codebase?

Reply:
Yes, the reason is that this is the target memory location for reading from TTree, so I use TTree's convention.